### PR TITLE
#493 Add support for custom oob redirect values

### PR DESCRIPTION
**Pull request by ctrombley**
cc @hadley 

For #493. This PR adds support for specifying a custom OOB parameter value when retrieving an authorization URL.  The Yahoo docs here show that their implementation requires a `redirect_uri` value of "oob":

https://developer.yahoo.com/oauth2/guide/flows_authcode/#step-2-get-an-authorization-url-and-authorize-access=

This PR also includes the following:
* An updated Yahoo OAuth2 demo
* An updated Yahoo base API URL
* Updated docs

### CODE REVIEW

File: R/oauth-init.R [hunk 1]
```diff
@@ -72,17 +74,20 @@ init_oauth2.0 <- function(endpoint, app, scope = NULL,
...
                           client_credentials = FALSE
                          ) {
 
   scope <- check_scope(scope)
-  use_oob <- check_oob(use_oob)
-
+  use_oob <- check_oob(use_oob, oob_value = NULL)
   if (use_oob) {
-    redirect_uri <- "urn:ietf:wg:oauth:2.0:oob"
+    redirect_uri <- if(!is.null(oob_value))
```
Comment (hadley) [inline R/oauth-init.R]: Please either write on one line, or use `{}`

File: R/oauth-init.R [hunk 2]
```diff
@@ -57,6 +57,8 @@ init_oauth1.0 <- function(endpoint, app, permission = NULL,
 #'   Otherwise, provide a URL to the user and prompt for a validation
 #'   code. Defaults to the of the \code{"httr_oob_default"} default,
 #'   or \code{TRUE} if \code{httpuv} is not installed.
+#' @param oob_value if provided, is used as a custom oob callback value.
```
Comment (hadley) [inline R/oauth-init.R]: Can you please make this description a bit more precise? It's not a callback; it's passed as a redirect parameter.

File: tests/testthat/test-oauth.R [hunk 3]
```diff
@@ -53,6 +53,30 @@ test_that("oauth_encode1 works", {
   expect_equal(orig_string, restored_string)
 })
 
+test_that("custom oob callback values can be provided", {
```
Comment (hadley) [inline tests/testthat/test-oauth.R]: TBH I don't think this is worth it; in httr2 we'll refactor to make this sort of testing easier, but for now I think it's better to omit as the test will be fragile.

Comment (ctrombley) [reply tests/testthat/test-oauth.R]: Removed.

File: R/oauth-init.R (hunk 1 above)
Comment (ctrombley) [reply R/oauth-init.R]: Done.

File: R/oauth-init.R (hunk 2 above)
Comment (ctrombley) [reply R/oauth-init.R]: Done.

File: R/oauth-init.R [hunk 4]
```diff
@@ -72,17 +75,17 @@ init_oauth2.0 <- function(endpoint, app, scope = NULL,
...
                           client_credentials = FALSE
                          ) {
 
   scope <- check_scope(scope)
-  use_oob <- check_oob(use_oob)
-
+  use_oob <- check_oob(use_oob, oob_value)
   if (use_oob) {
-    redirect_uri <- "urn:ietf:wg:oauth:2.0:oob"
+    redirect_uri <- ifelse(!is.null(oob_value), oob_value, "urn:ietf:wg:oauth:2.0:oob")
```
Comment (hadley) [inline R/oauth-init.R]:
```suggestion
    redirect_uri <- if (!is.null(oob_value)) oob_value else "urn:ietf:wg:oauth:2.0:oob"
```

`ifelse()` is specifically for creating vectors; you can just use an inline `if` here

**Comment by ctrombley** [comment]
Done. I updated the package to 1.8.2.  Let me know if there’s anything else I can help with!

File: DESCRIPTION [hunk 5]
```diff
@@ -1,6 +1,6 @@
 Package: httr
 Title: Tools for Working with URLs and HTTP
-Version: 1.3.1.9000
+Version: 1.3.2
```
Comment (hadley) [inline DESCRIPTION]: Please revert this change - the development version always ends in .9000 (and it's my job to increment when necessary)

Comment (ctrombley) [reply DESCRIPTION]: Done.  I also moved my bullets in NEWS.md under `1.3.1.9000`.

**Comment by hadley** [comment]
Thanks @ctrombley!
